### PR TITLE
Post mentions for participants and discussions

### PR DESCRIPTION
**Pull request by jasondellaluce**
Targets #53. Initial support for mentions in discussion's posts.

### CODE REVIEW

File: lib/chatham_app.dart [hunk 1]
```diff
@@ -226,6 +234,11 @@ class ChathamAppState extends State<ChathamApp>
               this.sendDeviceToServer(userDeviceRepository, state.me);
             }
           }),
+          BlocListener <DiscussionListBloc, DiscussionListState> (listener: (context, state) {
+            if (state is DiscussionListLoaded) {
```
Comment (jasondellaluce) [inline lib/chatham_app.dart:238]: Check the loading flag here

File: lib/data/provider/queries.dart [hunk 2]
```diff
@@ -60,6 +60,17 @@ const PostInfoFragment = """
...
+        title
+        anonymityType
+      }
+      ... on Participant {
+        isAnonymous
+        participantID
+      }
+    }
     quotedPost{
       ...QuotedPostInfoFragment
```
Comment (jasondellaluce) [inline lib/data/provider/queries.dart:75]: Add mentioned entities fragment on QuotedPostInfoFragment too

File: lib/widgets/input/delphis_input_mentions_popup.dart [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+
+  @override
+  _DelphisInputMentionsPopupState createState() => _DelphisInputMentionsPopupState();
+
+}
+
+class _DelphisInputMentionsPopupState extends State<DelphisInputMentionsPopup> {
+  final popupEntryHeight = 60.0;
+  final popupMaxEntries = 3;
+  final popupWidth = 320.0;
```
Comment (jasondellaluce) [inline lib/widgets/input/delphis_input_mentions_popup.dart:48]: Use mediaquery to adapt to screen width

File: lib/widgets/input/delphis_input_mentions_popup.dart [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+          discussion: discussions[index],
+          onTap: () {
+            discussionMentionAutoComplete(discussions[index]);
+          },
+        );
+      }
+    );
+  }
+
+  String getLastMentionAttempt(RegExp regex, String mentionSymbol) {
```
Comment (jasondellaluce) [inline lib/widgets/input/delphis_input_mentions_popup.dart:263]: Not needed anymore

File: lib/data/provider/queries.dart (hunk 2 above)
Comment (jasondellaluce) [reply lib/data/provider/queries.dart:75]: Done

File: lib/chatham_app.dart (hunk 1 above)
Comment (jasondellaluce) [reply lib/chatham_app.dart:238]: Done

File: lib/widgets/input/delphis_input_mentions_popup.dart (hunk 3 above)
Comment (jasondellaluce) [reply lib/widgets/input/delphis_input_mentions_popup.dart:48]: Done

File: lib/widgets/input/delphis_input_mentions_popup.dart (hunk 4 above)
Comment (jasondellaluce) [reply lib/widgets/input/delphis_input_mentions_popup.dart:263]: Done

File: lib/bloc/discussion_list/discussion_list_bloc.dart [hunk 5]
```diff
@@ -34,25 +34,23 @@ class DiscussionListBloc
         currentList = currentState.discussionList;
       }
 
-      yield DiscussionListLoaded(discussionList: currentList, isLoading: true);
+      yield DiscussionListLoaded(discussionList: currentList, isLoading: true, visibleDiscussionList: currentList);
       try {
+        var visibleList = await this.repository.getMyDiscussionList();
```
Comment (nedrocks) [inline lib/bloc/discussion_list/discussion_list_bloc.dart:39]: careful about this throwing (if user isn't logged in for instance.)

File: lib/chatham_app.dart (hunk 1 above)
Comment (nedrocks) [reply lib/chatham_app.dart:238]: If you have a better thought here on how to set up this state lmk

File: lib/screens/discussion/discussion_header.dart [hunk 6]
```diff
@@ -25,11 +25,20 @@ class DiscussionHeader extends StatefulWidget {
...
   void initState() {
     this.isMultilineTile = true;
     this.isFirstBuild = true;
+    this.isDisposed = false;
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    this.isDisposed = true;
```
Comment (nedrocks) [inline lib/screens/discussion/discussion_header.dart:40]: Use `mounted`

File: lib/widgets/delphis_rich_text/delphis_rich_text.dart [hunk 7]
```diff
@@ -0,0 +1,80 @@
+import 'package:flutter/gestures.dart';
```
Comment (nedrocks) [inline lib/widgets/delphis_rich_text/delphis_rich_text.dart:1]: Do you think we should add tests for this?

**Pull request merged**